test/elixir/test/search_partition_test.exs

```diff
@@ -139,4 +147,25 @@ defmodule SearchPartitionTest do
     assert ids == ["bar:1", "bar:5", "bar:9", "foo:2", "bar:3", "foo:4", "foo:6", "bar:7", "foo:8", "foo:10"]
   end
 
+  @tag :with_db
+  test "normal search on non-partitioned dbs with limit", context do
```

Comment:
Could you break this up into three tests. 1 test with value under limit. Another with no limit and another with a value over the limit.

Reply:
Good idea. I split this to 3 individual test cases.